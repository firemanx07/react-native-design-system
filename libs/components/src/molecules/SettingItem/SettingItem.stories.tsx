import { ComponentStory, ComponentMeta } from '@storybook/react';
import React from 'react';

import { StorybookScreen, StorybookSection } from '../../storybook';
import {
  SettingItem as SettingItemBase,
  PropsType as SettingItemProps,
} from './SettingItem';

const Story: ComponentMeta<typeof SettingItemBase> = {
  component: SettingItemBase,
  title: 'Molecules/Setting Item',
};
export default Story;

const Template: ComponentStory<typeof SettingItemBase> = (
  args: SettingItemProps,
) => (
  <StorybookScreen title="SettingItem:">
    <StorybookSection>
      <SettingItemBase {...args} />
    </StorybookSection>
    <StorybookSection>
      <SettingItemBase {...args} isActive />
    </StorybookSection>
  </StorybookScreen>
);

const parameters = {
  controls: {
    include: ['title', 'subtitle', 'disabled'],
  },
};

const defaultArgs = {
  title: 'Likes',
  subtitle: 'Someone liked your photos',
  onToggle: () => {
    //this is mock
  },
  isActive: false,
  disabled: false,
};

export const SettingItem = Template.bind({});
SettingItem.parameters = parameters;
SettingItem.args = {
  ...defaultArgs,
};
