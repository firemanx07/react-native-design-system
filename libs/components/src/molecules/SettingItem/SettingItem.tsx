import { styled } from '@proxym/themes';
import React, { memo } from 'react';
import { ViewStyle, StyleProp } from 'react-native';

import { BaseText, TextVariant, TextWeight, Toggle } from '../../atoms';
import { TestIDType } from '../../types';

export type PropsType = {
  title: string;
  onToggle: (isActive: boolean) => void;
  isActive: boolean;
  disabled?: boolean;
  subtitle?: string;
  style?: StyleProp<ViewStyle>;
} & TestIDType;

export const SettingItem = ({
  title,
  subtitle,
  onToggle,
  disabled,
  isActive,
  style,
  testID,
}: PropsType) => {
  return (
    <Container style={style} testID={testID}>
      <TextWrapper>
        <BaseText
          variant={TextVariant.H3}
          weight={TextWeight.Bold}
          numberOfLines={2}
        >
          {title}
        </BaseText>
        {!!subtitle && (
          <Subtitle
            variant={TextVariant.M}
            weight={TextWeight.SemiBold}
            numberOfLines={2}
          >
            {subtitle}
          </Subtitle>
        )}
      </TextWrapper>
      <Toggle isActive={isActive} onPress={onToggle} disabled={disabled} />
    </Container>
  );
};

const Container = styled.View`
  flex-direction: row;
  align-items: center;
  background-color: ${({ theme }) => theme.ds.colors.light};
  padding: ${({ theme }) => theme.ds.spacing.primary}px;
  min-height: 60px;
`;

const TextWrapper = styled.View`
  flex: 1;
  padding-right: ${({ theme }) => theme.ds.spacing.small}px;
`;

const Subtitle = styled(BaseText)`
  padding-top: ${({ theme }) => theme.ds.spacing.tiny}px;
  color: ${({ theme }) => theme.ds.colors.grayDark};
`;

export default memo(SettingItem);
