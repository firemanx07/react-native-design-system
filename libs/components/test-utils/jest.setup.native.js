import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();
require('react-native-gesture-handler/jestSetup.js');

console.warn = () => {
  //this is mock
};

Animated['timing'] = (value, config) => ({
  start: callback => {
    value.setValue(config.toValue);
    callback && callback({ finished: true });
  },
  stop: jest.fn(),
});
