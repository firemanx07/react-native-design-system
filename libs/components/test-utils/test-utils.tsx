import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
