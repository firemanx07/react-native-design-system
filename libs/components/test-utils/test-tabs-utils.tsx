import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
