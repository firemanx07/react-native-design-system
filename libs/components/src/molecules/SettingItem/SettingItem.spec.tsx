import { testAllStories } from '../../../test-utils';
import * as stories from './SettingItem.stories';

describe('SettingItem tests', () => {
  testAllStories(stories);
});
