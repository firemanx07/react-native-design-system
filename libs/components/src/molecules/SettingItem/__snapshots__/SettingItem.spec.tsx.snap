// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingItem tests SettingItem snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text
    fontFamily="Rubik-Bold"
    fontSize={20}
    lineHeight={26}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Bold",
          "fontSize": 20,
          "lineHeight": 26,
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    SettingItem:
  </Text>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flexDirection": "row",
            "minHeight": 60,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingRight": 8,
            },
          ]
        }
      >
        <Text
          fontFamily="Rubik-Bold"
          fontSize={16}
          lineHeight={20}
          numberOfLines={2}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-Bold",
                "fontSize": 16,
                "lineHeight": 20,
              },
            ]
          }
        >
          Likes
        </Text>
        <Text
          fontFamily="Rubik-SemiBold"
          fontSize={14}
          lineHeight={20}
          numberOfLines={2}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-SemiBold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#C7C7D2",
                "paddingTop": 4,
              },
            ]
          }
        >
          Someone liked your photos
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#DEDFE0",
            "borderRadius": 20,
            "height": 40,
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "width": 56,
          }
        }
      >
        <View
          collapsable={false}
          size={24}
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "elevation": 2,
              "height": 24,
              "shadowColor": "#2C2C2C",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "transform": [
                {
                  "translateX": 0,
                },
              ],
              "width": 24,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flexDirection": "row",
            "minHeight": 60,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingRight": 8,
            },
          ]
        }
      >
        <Text
          fontFamily="Rubik-Bold"
          fontSize={16}
          lineHeight={20}
          numberOfLines={2}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-Bold",
                "fontSize": 16,
                "lineHeight": 20,
              },
            ]
          }
        >
          Likes
        </Text>
        <Text
          fontFamily="Rubik-SemiBold"
          fontSize={14}
          lineHeight={20}
          numberOfLines={2}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-SemiBold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#C7C7D2",
                "paddingTop": 4,
              },
            ]
          }
        >
          Someone liked your photos
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#4D37D9",
            "borderRadius": 20,
            "height": 40,
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "width": 56,
          }
        }
      >
        <View
          collapsable={false}
          size={24}
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "elevation": 2,
              "height": 24,
              "shadowColor": "#2C2C2C",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "transform": [
                {
                  "translateX": 16,
                },
              ],
              "width": 24,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
